### PR TITLE
Setup golangci-lint

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

fixes of https://github.com/containernetworking/plugins/issues/663

### CODE REVIEW

**Comment by pigletfly** [comment]
@squeed @dcbw can you help to trigger the Github workflow ?

**Comment by squeed** [comment]
wow, thanks!

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - gofmt
+    - goimports
+    - revive
+    - ineffassign
+    - vet
+    - unused
+    - misspell
+    - gofumpt
+  disable:
+    - errcheck
```
Comment (squeed) [inline .golangci.yml:16]: yeah, we should eventually enable this too :)

**Review by squeed — APPROVED** [review]

**Comment by squeed** [comment]
Fixes #663

File: .golangci.yml (hunk 1 above)
Comment (pigletfly) [reply .golangci.yml:16]: yeah, if we need more more linters, just add them in .golangci.yml.

**Comment by pigletfly** [comment]
change lint event trigger to 
```
  pull_request:
    types: ['opened', 'edited', 'reopened', 'synchronize']
```

**Comment by pigletfly** [comment]
@squeed another trigger pls

**Comment by pigletfly** [comment]
@squeed another trigger pls

**Comment by pigletfly** [comment]
> @squeed another trigger pls

let's have another try

**Comment by dcbw** [comment]
@pigletfly Existing bits LGTM; just a couple more to fix. Would you be willing to finish them up?

Thanks! Awesome work.

**Comment by pigletfly** [comment]
> @pigletfly Existing bits LGTM; just a couple more to fix. Would you be willing to finish them up?
> 
> Thanks! Awesome work.

OK, it's weird why the lint errors didn't appear on my laptop.

**Comment by squeed** [comment]
argh there's a merge conflict

**Pull request closed without merging**